src/bulk/common.ts

```diff
@@ -641,205 +663,202 @@ class BulkWriteError extends MongoError {
    */
   constructor(bulkOperation: any) {
     this.s = bulkOperation.s;
   }
 
-  /**
-   * Add a multiple update operation to the bulk operation
-   *
-   * @function
-   * @param updateDocument An update field for an update operation. See {@link https://docs.mongodb.com/manual/reference/command/update/#update-command-u u documentation}
-   * @param reference for more information.
-   * @throws MongoError If operation cannot be added to bulk write
-   * @returns {any} A reference to the parent BulkOperation
-   */
-  update(updateDocument: any): any {
+  /** Add a multiple update operation to the bulk operation */
+  update(updateDocument: DocumentWithHint) {
     // Perform upsert
-    const upsert = typeof this.s.currentOp.upsert === 'boolean' ? this.s.currentOp.upsert : false;
+    const upsert = typeof this.s.currentOp?.upsert === 'boolean' ? this.s.currentOp.upsert : false;
 
     // Establish the update command
     const document = {
-      q: this.s.currentOp.selector,
+      q: this.s.currentOp?.selector,
       u: updateDocument,
       multi: true,
-      upsert: upsert
-    } as any;
-
-    if (updateDocument.hint) {
-      document.hint = updateDocument.hint;
-    }
+      upsert: upsert,
+      ...(updateDocument.hint ? { hint: updateDocument.hint } : {})
+    };
 
-    // Clear out current Op
-    this.s.currentOp = null;
+    this.clearOutCurrentOp();
     return this.s.options.addToOperationsList(this, UPDATE, document);
   }
 
-  /**
-   * Add a single update operation to the bulk operation
-   *
-   * @function
-   * @param updateDocument An update field for an update operation. See {@link https://docs.mongodb.com/manual/reference/command/update/#update-command-u u documentation}
-   * @param reference for more information.
-   * @throws {MongoError} If operation cannot be added to bulk write
-   * @returns {any} A reference to the parent BulkOperation
-   */
-  updateOne(updateDocument: any): any {
+  clearOutCurrentOp() {
+    this.s.currentOp = undefined;
+  }
+
+  /** Add a single update operation to the bulk operation */
+  updateOne(updateDocument: DocumentWithHint): any {
     // Perform upsert
-    const upsert = typeof this.s.currentOp.upsert === 'boolean' ? this.s.currentOp.upsert : false;
+    const upsert = typeof this.s.currentOp?.upsert === 'boolean' ? this.s.currentOp.upsert : false;
 
     // Establish the update command
     const document = {
-      q: this.s.currentOp.selector,
+      q: this.s.currentOp?.selector,
       u: updateDocument,
       multi: false,
-      upsert: upsert
-    } as any;
-
-    if (updateDocument.hint) {
-      document.hint = updateDocument.hint;
-    }
+      upsert: upsert,
+      ...(updateDocument.hint ? { hint: updateDocument.hint } : {})
+    };
 
     if (!hasAtomicOperators(updateDocument)) {
       throw new TypeError('Update document requires atomic operators');
     }
 
-    // Clear out current Op
-    this.s.currentOp = null;
+    this.clearOutCurrentOp();
```

Comment:
Was this function added for readability? if so, my personal nit, maybe just `clearCurrentOp`?